src/app/guards/user.guard.ts

```diff
@@ -0,0 +1,26 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from './../services/auth.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserGuard implements CanActivate {
+
+  constructor(
+    private auth: AuthService,
+    private router: Router
+  ) {}
+
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
+      if (this.auth.isLoggedIn) {
+        this.router.navigate(['my-account']);

```

Comment:
Can we pass my-account or login as a parameter and avoid two different services

Reply:
I will check once